src/dex/synthetix/synthetix-state.ts

```diff
@@ -0,0 +1,933 @@
+      this.onchainConfigValues.aggregatorsAddresses,
+    ).reduce<Record<string, Address>>((acc, curr) => {
+      const [key, address] = curr;
+      if (address !== NULL_ADDRESS) {
+        acc[key] = address;
+      }
+      return acc;
+    }, {});
+
+    const [packCounter, slot0TickCumulativesSuspensionsLatestRoundCallData] =
+      this._buildObserveSlot0SuspensionsLatestRoundCallData(
+        addressesFromPK,
+        aggregatorAddressesWithoutZeros,
+      );
+
+    const slot0TickCumulativesSuspensionsLatestRound = (
+      await this.multiWrapper.tryAggregate<
+        Record<0 | 1, bigint> | Slot0 | boolean[] | boolean | LatestRoundData
+      >(true, slot0TickCumulativesSuspensionsLatestRoundCallData, blockNumber)
+    ).map(d => d.returnData) as (
+      | Record<0 | 1, bigint>
+      | boolean
+      | boolean[]
+      | LatestRoundData
+    )[];
+
+    let addressesFromPKBoundary = addressesFromPK.length * packCounter;
+    const slot0AndTickCumulatives =
+      slot0TickCumulativesSuspensionsLatestRound.slice(
+        0,
+        addressesFromPKBoundary,
+      );
+
+    const suspensions = slot0TickCumulativesSuspensionsLatestRound.slice(
+      addressesFromPKBoundary,
+      // Number of suspension requests
+      addressesFromPKBoundary + 3,
+    );
+
+    const latestRoundDatas = slot0TickCumulativesSuspensionsLatestRound.slice(
+      addressesFromPKBoundary + 3,
+    ) as LatestRoundData[];
+
+    const isSystemSuspended = suspensions[0] as boolean;
+    const synthSuspensions = suspensions[1] as boolean[];
+    const synthExchangeSuspensions = suspensions[2] as boolean[];
+
+    const areSynthsSuspended = Object.keys(
+      this.onchainConfigValues.addressToKey,
+    ).reduce<Record<string, boolean>>((acc, curr, i) => {
+      acc[curr] =
+        synthSuspensions[i] === true || synthExchangeSuspensions[i] === true;
+      return acc;
+    }, {});
+
+    const uniswapV3Slot0: Record<Address, Slot0> = {};
+    const tickCumulatives: Record<Address, Record<0 | 1, bigint>> = {};
+
+    _.chunk(slot0AndTickCumulatives, packCounter).forEach((result, i) => {
+      const address = addressesFromPK[i];
+      const [tickCumulative, slot0] = result as [Record<0 | 1, bigint>, Slot0];
+
+      uniswapV3Slot0[address] = slot0;
+      tickCumulatives[address] = tickCumulative;
+    });
+
+    const [_packCounter, observationsRoundAndOverriddenCallData] =
+      this._buildObservationsRoundAndOverriddenCallData(
+        addressesFromPK,
+        uniswapV3Slot0,
+        aggregatorAddressesWithoutZeros,
+        latestRoundDatas,
+      );
+
+    const [block, observationsRoundDataAndOverridden] = await Promise.all([
+      this.dexHelper.web3Provider.eth.getBlock(blockNumber || 'latest'),
+      this.multiWrapper.tryAggregate<
+        OracleObservation | LatestRoundData | string
+      >(true, observationsRoundAndOverriddenCallData, blockNumber),
+    ]);
+
+    addressesFromPKBoundary = addressesFromPK.length * packCounter;
```

Comment:
I think this should be `_packCounter` not `packCounter` (turns out they are the same currently, so it happens to work).

Reply:
Good catch, thanks! Fixed it 👍